python/lsst/pipe/base/_dataset_handle.py

```diff
@@ -0,0 +1,153 @@
+# This file is part of pipe_base.
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+from __future__ import annotations
+
+__all__ = ["InMemoryDatasetHandle"]
+
+import dataclasses
+from typing import Any, Optional
+
+from lsst.daf.butler import DataCoordinate, DimensionUniverse, StorageClass, StorageClassFactory
+from lsst.utils.introspection import get_full_type_name
+
+# Use an empty dataID as a default.
+_DEFAULT_DATAID = DataCoordinate.standardize({}, universe=DimensionUniverse())
```

Comment:
I'm not a fan of instantiating the default dimension universe at import time unconditionally.  Mind turning this into a function, and using the `default_factory` argument to `dataclasses.field` instead?

There's also a `DataCoordinate.makeEmpty(universe)` method you can use.

And if we're not absolutely certain we need a default for the data ID, it might be best to remove it to avoid surprising behavior when some task tries to look at the data ID.